baseapp/baseapp.go

```diff
@@ -48,9 +48,10 @@ type BaseApp struct {
 	// See methods setCheckState and setDeliverState.
 	// .valUpdates accumulate in DeliverTx and are reset in BeginBlock.
 	// QUESTION: should we put valUpdates in the deliverState.ctx?
-	checkState   *state           // for CheckTx
-	deliverState *state           // for DeliverTx
-	valUpdates   []abci.Validator // cached validator changes from DeliverTx
+	checkState       *state           // for CheckTx
+	deliverState     *state           // for DeliverTx
+	valUpdates       []abci.Validator // cached validator changes from DeliverTx
+	absentValidators []int32          // absent validators from begin block
```

Comment:
do we really need these in the context ? I would think we can just use them where necessary in BeginBlock - no ?

Reply:
hmm okay so then we just include the fee distribution calculation section to be used in begin block and then can avoid passing it around in the context... think this is possible